notarization_poller/src/notarization_poller/task.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/env python
+"""Task execution.
+
+Attributes:
+    NOTARIZATION_POLL_REGEX: the regex to find the notarization status
+    Task (object): the task object
+    log (logging.Logger): the log object for the module
+
+"""
+
+import asyncio
+import gzip
+import logging
+import os
+import pprint
+import re
+import signal
+import traceback
+
+import aiohttp
+import arrow
+import async_timeout
+import taskcluster
+import taskcluster.exceptions
+from taskcluster.aio import Queue
+from taskcluster.client import createTemporaryCredentials
+
+from notarization_poller.constants import get_reversed_statuses
+from notarization_poller.exceptions import RetryError
+from scriptworker_client.aio import download_file, retry_async
+from scriptworker_client.constants import STATUSES
+from scriptworker_client.exceptions import Download404, DownloadError, TaskError
+from scriptworker_client.utils import load_json_or_yaml, makedirs, rm, run_command
+
+log = logging.getLogger(__name__)
+NOTARIZATION_POLL_REGEX = re.compile(r"Status: (?P<status>success|invalid)")
+
+
+class Task:
+    """Manages all information related to a single running task."""
+
+    reclaim_fut = None
+    task_fut = None
+    complete = False
+    process = None
+    uuids = None
+
+    def __init__(self, config, claim_task, event_loop=None):
+        """Initialize Task."""
+        self.config = config
+        self.task_id = claim_task["status"]["taskId"]
+        self.run_id = claim_task["runId"]
+        self.claim_task = claim_task
+        self.event_loop = event_loop or asyncio.get_event_loop()
+        self.task_dir = os.path.join(config["work_dir"], self.task_id)
+        self.log_path = os.path.join(self.task_dir, "live_backing.log")
+        self.poll_log_path = os.path.join(self.task_dir, "polling.log")
+        rm(self.task_dir)
+        makedirs(self.task_dir)
+        self.reclaim_fut = event_loop.create_task(self.reclaim_task())
+        self._reclaim_task = {}
+        self.task_fut = event_loop.create_task(self.run_task())
+
+    @property
+    def task_credentials(self):
+        """Return the temporary credentials returned from [re]claimWork."""
+        return self._reclaim_task.get("credentials", self.claim_task["credentials"])
```

Comment:
I avoided creating a session and queue and passing it around, because it took a lot of effort in scriptworker to do so, and I'm not sure we have a clear win. Aiohttp wants to force us to use `async with aiohttp.ClientSession()` which means a long lived generator or having to hack around those efforts (we took the latter approach in scriptworker, and have had to update our hacks every time aiohttp changes their rules).